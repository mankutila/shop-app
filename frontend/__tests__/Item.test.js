import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'uniqueId',
  title: 'One more item',
  description: 'One more item description',
  price: 5000,
  image: 'image.jpg',
  largeImage: 'large-image.jpg',
};

describe('Item component', function () {
  it('should render and match snapshot', function () {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  /*it('should render', function () {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find('PriceTag');
    const img = wrapper.find('img');

    expect(PriceTag.dive().text()).toEqual('BYNÂ 50');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);

  });

  it('should render the buttons properly', function () {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');
    console.log(buttonList.debug());
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('AddToCart')).toHaveLength(1);
    expect(buttonList.find('DeleteItem')).toHaveLength(1);

  });*/
});
