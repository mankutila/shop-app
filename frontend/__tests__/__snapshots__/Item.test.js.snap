// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should render and match snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="One more item"
    src="image.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "uniqueId",
          },
        }
      }
    >
      <a>
        One more item
      </a>
    </Link>
  </Title>
  <PriceTag>
    BYNÂ 50
  </PriceTag>
  <p>
    One more item description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "uniqueId",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="uniqueId"
    />
    <DeleteItem
      id="uniqueId"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
